app/src/main/java/com/example/demouniverload/search/presenter/SearchPresenter.java

```diff
@@ -13,37 +12,36 @@
 import io.reactivex.disposables.Disposable;
 import io.reactivex.schedulers.Schedulers;
 
-public class SearchPresenter extends SearchFabric implements SearchContract.ISearchPresenter {
+public class SearchPresenter extends SearchFactory implements SearchContract.ISearchPresenter {
 
     private SearchContract.ISearchView searchView;
     private CompositeDisposable compositeDisposable = new CompositeDisposable();
+    private MethodsFactory methodsFactory;
 
-    public SearchPresenter(SearchView searchView) {
-
+    public SearchPresenter(SearchView searchView, MethodsFactory methodsFactory) {
```

Comment:
You can send `YandexRequester` and `YandexParser` directly into `SearchPresenter` constructor.
```suggestion
    public SearchPresenter(SearchView searchView, YendexParser parser, YandexRequester requester) {
```